app/grants/clr.py

```diff
@@ -571,3 +578,51 @@ def predict_clr_live(grant, contributor, amount, is_postive_vote=True):
     )
 
     return predicted_clr_match
+
+
+'''
+    Server Side Grants Transaction Validator
+
+    Args:
+        list_contributions : csv | text | file
+        
+    Returns:
+        status: sucess or fail, reason
+        sucess_validator: [{'sucess': <int>, 'reason': <Object>}]
+        fail_validator: [{'fail': <int>, 'reason': <Object>}]
+
+'''
+
+def grants_transaction_validator(list_contributions):
+    """This function check grants transaction list"""
+
+    df = pd.read_csv(list_contributions, sep=" ")
+    check_transaction = lambda txid: w3.eth.getTransaction(txid)
+    check_amount = lambda amount: int(amount[75:], 16) if len(amount) == 138 else print (f"{bcolors.FAIL}{bcolors.UNDERLINE}{index_transaction}{txid[:10]} -> status: 0 False - amount was off by 0.001{bcolors.ENDC}")
+    check_token = lambda token_address: len(token_address) == 42
+
+    for index_transaction, txid in enumerate(df.txid):
+        try:
+            transaction = check_transaction(txid)
+            amount =  check_amount(transaction.input)
```

Comment:
does this handle ERC20 transactions AND eth transactions?

Reply:
yeah 👍